### PR TITLE
Move delete deploy keys into service layer

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: services/asymkey/deploy_key.go [hunk 1]
```diff
@@ -19,7 +79,19 @@ func DeleteDeployKey(ctx context.Context, doer *user_model.User, id int64) error
...
-	if err := models.DeleteDeployKey(dbCtx, doer, id); err != nil {
+	key, err := asymkey_model.GetDeployKeyByID(ctx, id)
+	if err != nil {
+		if asymkey_model.IsErrDeployKeyNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("GetDeployKeyByID: %w", err)
+	}
+
+	if err := checkDeployPerm(ctx, doer, key.RepoID, key.ID); err != nil {
```
Comment (Zettat123) [inline services/asymkey/deploy_key.go]: It looks like the permission is checked outside of `DeleteDeployKey`. Do we need to check the permissions again using `checkDeployPerm`?

Comment (lunny) [reply services/asymkey/deploy_key.go]: The function removed and only check whether the key belongs to the repository.

File: services/asymkey/deploy_key.go [hunk 2]
```diff
@@ -5,21 +5,70 @@ package asymkey
...
 
-	"code.gitea.io/gitea/models"
+	asymkey_model "code.gitea.io/gitea/models/asymkey"
 	"code.gitea.io/gitea/models/db"
+	repo_model "code.gitea.io/gitea/models/repo"
 	user_model "code.gitea.io/gitea/models/user"
 )
 
+// DeleteRepoDeployKeys deletes all deploy keys of a repository. permissions check should be done outside
+func DeleteRepoDeployKeys(ctx context.Context, doer *user_model.User, repoID int64) (int, error) {
```
Comment (Zettat123) [inline services/asymkey/deploy_key.go]: Maybe we should also remove the `doer` parameter

Comment (lunny) [reply services/asymkey/deploy_key.go]: Done.

**Review by Zettat123 — APPROVED** [review]

File: models/repo.go [hunk 3]
```diff
@@ -315,48 +312,3 @@ func DoctorUserStarNum(ctx context.Context) (err error) {
...
-	key, err := asymkey_model.GetDeployKeyByID(ctx, id)
-	if err != nil {
-		if asymkey_model.IsErrDeployKeyNotExist(err) {
-			return nil
-		}
-		return fmt.Errorf("GetDeployKeyByID: %w", err)
-	}
-
-	// Check if user has access to delete this key.
-	if !doer.IsAdmin {
```
Comment (delvh) [inline models/repo.go:330]: What happened to all these permission checks?
I don't see them in the new code anymore…

Comment (lunny) [reply models/repo.go:330]: These checks are duplicated because in routers layer, reqAdmin has been used.

**Comment by lunny** [comment]
Last call for another review @go-gitea/technical-oversight-committee

**Review by delvh — APPROVED** [review]

**Pull request merged**